source/vulkan/vulkan_hooks.cpp

```diff
@@ -301,12 +479,28 @@ VkResult VKAPI_CALL vkCreateDevice(VkPhysicalDevice physicalDevice, const VkDevi
 	dispatch_table.CmdSetViewport = (PFN_vkCmdSetViewport)get_device_proc(device, "vkCmdSetViewport");
 	dispatch_table.CmdDraw = (PFN_vkCmdDraw)get_device_proc(device, "vkCmdDraw");
 	dispatch_table.CmdDrawIndexed = (PFN_vkCmdDrawIndexed)get_device_proc(device, "vkCmdDrawIndexed");
+	dispatch_table.CmdDrawIndexedIndirect = (PFN_vkCmdDrawIndexedIndirect)get_device_proc(device, "vkCmdDrawIndexedIndirect");
 	dispatch_table.CmdPipelineBarrier = (PFN_vkCmdPipelineBarrier)get_device_proc(device, "vkCmdPipelineBarrier");
 	dispatch_table.DebugMarkerSetObjectNameEXT = (PFN_vkDebugMarkerSetObjectNameEXT)get_device_proc(device, "vkDebugMarkerSetObjectNameEXT");
 
 	{ const std::lock_guard<std::mutex> lock(s_mutex);
 		s_device_mapping[device] = physicalDevice;
 		s_device_dispatch[get_dispatch_key(device)] = dispatch_table;
+		reshade::vulkan::draw_call_tracker device_tracker;
+		s_device_tracker[device] = device_tracker;
+	}
+
+	// retrieve the associated command queues
+	for (uint32_t i = 0; i < pCreateInfo->queueCreateInfoCount; i++)
+	{
+		VkDeviceQueueCreateInfo queueInfo = pCreateInfo->pQueueCreateInfos[i];
+
+		for (uint32_t j = 0; j < queueInfo.queueCount; j++)
+		{
+			VkQueue queue = VK_NULL_HANDLE;
+			vkGetDeviceQueue(device, queueInfo.queueFamilyIndex, j, &queue);
```

Comment:
It is not valid to call Vulkan functions directly. They need to go through the dispatch tables.

Reply:
I guess it is fixed now